data/preprocess/util.py

```diff
@@ -252,10 +256,18 @@ def store_map(symbol_map: Dict, filename: str):
         for symbol, index in symbol_map.items():
             f.write(f"{index}\t{symbol}\n")
 
+def dump_buffer_to_file(buffer, fptr):
+    """Dump buffer to file."""
+    fptr.write(f"{buffer}")
 
-def write_triple(f, ent, rel, t, S, P, O):
+def write_triple(f, buffer, ent, rel, t, S, P, O):
     """Write a triple to a file. """
-    f.write(str(ent[t[S]]) + "\t" + str(rel[t[P]]) + "\t" + str(ent[t[O]]) + "\n")
+    buffer += f"{str(ent[t[S]])}\t{str(rel[t[P]])}\t{str(ent[t[O]])}\n"
+    if len(buffer) > BUFFER_SIZE:
+        dump_buffer_to_file(buffer, f)
```

Comment:
did you make sure, that the rest of the buffer is written to file at every necessary position? you did so in the RawSplit but process_triple is called also in other splits